distribution.pm

```diff
@@ -107,7 +107,7 @@ sub become_root {
 
 =head2 script_run
 
-  script_run($cmd [, timeout => $timeout] [, output => $output] [,quiet => $quiet])
+  script_run($cmd [, timeout => $timeout] [, output => $output] [,quiet => $quiet] [,max_interval => $max_interval])
```

Comment:
@SergioAtSUSE might this line interfere and overwrite the default? I cant give all the arguments as some are not meant for type_string

Reply:
Well this is just the comment.

I assume you mean line 143? Where you are passing a maybe empty (undef) `max_interval` key?

It doesn't, as the output of the pasted script demonstrate. `if ($args{max_interval})` checks is a value for that key exists, not if the key exists.

Or do you mean something else?

Reply:
If you mean that you should only pass the parameters that are expected for type_string, I agree.
I am not sure how much benefit of "spare typing" vs "maintenance" we gain passing a hash instead of only the values we need.
Because removing the keys from `@args` may not be a good idea if some logic later needs to use them.

Reply:
I meant passing an undef value explicitly, but if you tested it it should be fine.
Also I agree on not manipulating the hash just to pass the complete hash that contains the same as only the value we want to pass

Reply:
So, parameters in general are somehow being mangled, https://travis-ci.org/github/os-autoinst/os-autoinst/builds/694623878#L866 (but at least seems to be consistent now: https://travis-ci.org/github/os-autoinst/os-autoinst/builds/694623878#L915)